compiler/src/iree/compiler/InputConversion/Common/QuantizedMatmulToMatmul.cpp

```diff
@@ -28,51 +28,77 @@ bool isConstantZero(Value val) {
   return constIntOp && constIntOp.value() == 0;
 }
 
-// Pattern lowering quantized_matmul to matmul.
-// Always succeeds.
-//
+// Pattern lowering quantized_matmul to matmul and quantized_batch_matmul to
+// batch_matmul op.
 // This is implementing the math explained in Section 2.3 of
 // https://arxiv.org/abs/1712.05877.
 struct QuantizedMatmulToMatmul
-    : public OpRewritePattern<linalg::QuantizedMatmulOp> {
-  using OpRewritePattern<linalg::QuantizedMatmulOp>::OpRewritePattern;
+    : public OpInterfaceRewritePattern<linalg::LinalgOp> {
+  using OpInterfaceRewritePattern<linalg::LinalgOp>::OpInterfaceRewritePattern;
 
-  LogicalResult matchAndRewrite(linalg::QuantizedMatmulOp quantizedMatmulOp,
+  LogicalResult matchAndRewrite(linalg::LinalgOp op,
                                 PatternRewriter &rewriter) const override {
-    Location loc = quantizedMatmulOp.getLoc();
+    Location loc = op.getLoc();
+    ValueRange inputs{}, outputs{};
+    bool hasBatchDim = false;
+    if (auto qMatmulOp = dyn_cast<linalg::QuantizedMatmulOp>(&op)) {
+      inputs = qMatmulOp->getInputs();
+      outputs = qMatmulOp->getOutputs();
+    } else if (auto qBMatmulOp =
+                   dyn_cast<linalg::QuantizedBatchMatmulOp>(&op)) {
```

Comment:
Curious, is the `&` necessary here?

Reply:
It's not required. Thanks for pointing this out.